File: Golang/676. Implement Magic Dictionary.go
type MagicDictionary struct {
    wordDict map[int][]string
}


/** Initialize your data structure here. */
func Constructor() MagicDictionary {
    return MagicDictionary{wordDict: make(map[int][]string, 100)}
}


/** Build a dictionary through a list of words */
func (this *MagicDictionary) BuildDict(dict []string)  {
    for _, word := range dict {
        this.wordDict[len(word)] = append(this.wordDict[len(word)], word)
    }
}


/** Returns if there is any word in the trie that equals to the given word after modifying exactly one character */
func (this *MagicDictionary) Search(word string) bool {
    for _, item := range this.wordDict[len(word)] {
        count := 0
        for index, _ := range word {
            if item[index] != word[index] {
                count++
            }
        }
        if count == 1 {
            return true
        }
    }
    return false
}


/**
 * Your MagicDictionary object will be instantiated and called as such:
 * obj := Constructor();
 * obj.BuildDict(dict);
 * param_2 := obj.Search(word);
 */

File: Python3/676. Implement Magic Dictionary.py
class MagicDictionary:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.Dict = []
        

    def buildDict(self, dict):
        """
        Build a dictionary through a list of words
        :type dict: List[str]
        :rtype: void
        """
        self.Dict = dict
        

    def search(self, word):
        """
        Returns if there is any word in the trie that equals to the given word after modifying exactly one character
        :type word: str
        :rtype: bool
        """
        for item in self.Dict:
            if len(word) == len(item):
                count = 0
                for i in range(len(word)):
                    if word[i] != item[i]:
                        count += 1
                    if count > 1:
                        break
                if count == 1:
                    return True
        return False
        


# Your MagicDictionary object will be instantiated and called as such:
# obj = MagicDictionary()
# obj.buildDict(dict)
# param_2 = obj.search(word)